### PR TITLE
Epmlstrtai 112

### PR DESCRIPTION
**Pull request by SergeiLukianov**
step entity added, case & suit updated. Dozer mapper added and configured

### CODE REVIEW

File: Java/JDI/jdi-cucumber-test-generator/pom.xml [hunk 1]
```diff
@@ -113,6 +113,12 @@
             <artifactId>spring-test</artifactId>
             <version>4.3.10.RELEASE</version>
         </dependency>
+        <!--DTO to Entity mapping-->
+        <dependency>
+            <groupId>net.sf.dozer</groupId>
+            <artifactId>dozer</artifactId>
+            <version>5.5.1</version>
```
Comment (EvilXa0S) [inline Java/JDI/jdi-cucumber-test-generator/pom.xml:117]: Extract version into parent pom dependency mangement section

File: Java/JDI/jdi-cucumber-test-generator/src/main/java/com/epam/test_generator/DozerMapper.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        List cfg = new ArrayList();
+        cfg.add("dozerMapping.xml");
+        mapper = new DozerBeanMapper(cfg);
+    }
+
+    public static void map(Object source, Object dest) {
+        mapper.map(dest, source);
+    }
+
+    public static void main(String[] args) {
```
Comment (EvilXa0S) [inline Java/JDI/jdi-cucumber-test-generator/src/main/java/com/epam/test_generator/DozerMapper.java:23]: Do we really need ti method in our application? I doubt, please remove.

File: Java/JDI/jdi-cucumber-test-generator/src/main/java/com/epam/test_generator/config/DatabaseConfig.java [hunk 3]
```diff
@@ -1,16 +1,12 @@
 package com.epam.test_generator.config;
 
-//import org.apache.commons.dbcp2.BasicDataSource;
```
Comment (EvilXa0S) [inline Java/JDI/jdi-cucumber-test-generator/src/main/java/com/epam/test_generator/config/DatabaseConfig.java:3]: Please delete unused imports.

File: Java/JDI/jdi-cucumber-test-generator/src/main/java/com/epam/test_generator/dao/MockDao.java [hunk 4]
```diff
@@ -21,19 +21,19 @@
...
-                    "\tGiven User is on Home Page\n" +
-                    "\tWhen User Navigate to LogIn Page\n" +
-                    "\tAnd User enters UserName and Password\n" +
-                    "\tThen Message displayed Login Successfully\n" +
-                    " \n" +
-                    "Scenario: Successful LogOut\n" +
-                    "\tWhen User LogOut from the Application\n" +
-                    "\tThen Message displayed LogOut Successfully", null));
-        }
+//        for (int i = 0; i < 30; i++) {
```
Comment (EvilXa0S) [inline Java/JDI/jdi-cucumber-test-generator/src/main/java/com/epam/test_generator/dao/MockDao.java:24]: Please delete unused code

File: Java/JDI/jdi-cucumber-test-generator/src/main/java/com/epam/test_generator/entities/StepType.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
+package com.epam.test_generator.entities;
+
+public enum StepType {
+
```
Comment (pomkine) [inline Java/JDI/jdi-cucumber-test-generator/src/main/java/com/epam/test_generator/entities/StepType.java]: The same thing can be done like this:
```
public enum StepType {
 private String typeName;
 public StepType(String typeName){
  this.typeName = typeName;
}
 GIVEN("Given")
...
}
```
Doing this way it's less code to write :)

File: Java/JDI/jdi-cucumber-test-generator/src/main/java/com/epam/test_generator/entities/StepType.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    GIVEN {
```
Comment (EvilXa0S) [inline Java/JDI/jdi-cucumber-test-generator/src/main/java/com/epam/test_generator/entities/StepType.java]: Rewrite with something like 

GIVEN("given"), WHEN("When"),.....
private String stepTypeName;

StepType(String str) {
stepTypeName = str;
}

getStepTypeName(){
 return stepTypeName;
}
}

File: Java/JDI/jdi-cucumber-test-generator/src/main/java/com/epam/test_generator/entities/Suit.java [hunk 7]
```diff
@@ -18,18 +19,27 @@
...
     private String description;
 
-    @JsonManagedReference
+    private Integer priority;
+
+    private Date creationDate;
+
+    private String tags;
+
+    //    @JsonManagedReference
```
Comment (EvilXa0S) [inline Java/JDI/jdi-cucumber-test-generator/src/main/java/com/epam/test_generator/entities/Suit.java]: Do not keep commented code. Please delete.

File: Java/JDI/jdi-cucumber-test-generator/src/main/java/com/epam/test_generator/dto/CaseDTO.java [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+public class CaseDTO {
+    private Long id;
+
+    private String description;
+
+    @JsonManagedReference
+    private List<Step> steps;
+
+    @JsonBackReference
+    private Suit suit;
```
Comment (pomkine) [inline Java/JDI/jdi-cucumber-test-generator/src/main/java/com/epam/test_generator/dto/CaseDTO.java]: you shouldn't have references to hibernate entities in DTOs. This one should be DTO as well

File: Java/JDI/jdi-cucumber-test-generator/src/main/java/com/epam/test_generator/services/CaseService.java [hunk 9]
```diff
@@ -9,6 +9,7 @@
 
 import java.util.List;
 
+@Transactional
```
Comment (EvilXa0S) [inline Java/JDI/jdi-cucumber-test-generator/src/main/java/com/epam/test_generator/services/CaseService.java:12]: Why deleteing this from methods and putting only to class? Any specific purpose?

File: Java/JDI/jdi-cucumber-test-generator/src/main/resources/dozerMapping.xml [hunk 10]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<mappings xmlns="http://dozer.sourceforge.net" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://dozer.sourceforge.net
+        http://dozer.soundforge.net/schema/beanmapping.xsd">
+    <mapping>
```
Comment (EvilXa0S) [inline Java/JDI/jdi-cucumber-test-generator/src/main/resources/dozerMapping.xml]: Is this config really needed?

File: Java/JDI/jdi-cucumber-test-generator/src/test/java/com/epam/test_generator/controllers/CaseControllerTests.java [hunk 11]
```diff
@@ -14,30 +14,30 @@
...
-        caseController.addCaseToSuit(3, case3);
-    }
-
-    @Test
-    public void test2() {
-        Case case1 = new Case((long) 1, "new description", "", suitService.getSuit((long) 1));
-        //caseController.updateCase(case1);
-    }
+//
+//    @BeforeClass
```
Comment (EvilXa0S) [inline Java/JDI/jdi-cucumber-test-generator/src/test/java/com/epam/test_generator/controllers/CaseControllerTests.java]: Why tests are commented? Either delete (which better should not be done) or update for them to be green.

**Comment by EvilXa0S** [comment]
Resolve the conflicts!!!

File: Java/JDI/jdi-cucumber-test-generator/src/main/java/com/epam/test_generator/services/CaseService.java (hunk 9 above)
Comment (pomkine) [reply Java/JDI/jdi-cucumber-test-generator/src/main/java/com/epam/test_generator/services/CaseService.java:12]: @EvilXa0S  If you mark a class with @Transactional, all methods of that class become @Transactional.

**Pull request merged**